pkg/applicationserver/io/applicationpackages/grpc.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applicationpackages
+
+import (
+	"context"
+
+	pbtypes "github.com/gogo/protobuf/types"
+	"go.thethings.network/lorawan-stack/pkg/auth/rights"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+// appendImplicitAssociationsGetPaths appends implicit ttnpb.ApplicationPackageAssociation get paths to paths.
+func appendImplicitAssociationsGetPaths(paths ...string) []string {
+	return append(append(make([]string, 0, 1+len(paths)),
+		"package_name",
+	), paths...)
+}
+
+// GetPackages implements ttnpb.ApplicationPackageRegistryServer.
+func (s *server) GetPackages(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers) (*ttnpb.ApplicationPackages, error) {
+	if err := rights.RequireApplication(ctx, ids.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_INFO); err != nil {
```

Comment:
We're mixing a lot of rights here; info, settings, traffic read/write

I see it's analog with pub/sub, but that is really about traffic flow. I would say that packages define AS behavior and requires (only) basic settings rights to manage

@htdvisser thoughts?

Reply:
I think we should define a new right for managing packages and assignments.

Reading/writing the config of a package association can have very serious implications (you have a lot of power if you can read/write the config of the FOTA package)

Reply:
Fine with me. So `SETTINGS_PACKAGES`?